--- openhpi-3.8.0/plugins/ov_rest/ov_rest_re_discover.c.orig	2018-03-08 19:10:16.000000000 +0100
+++ openhpi-3.8.0/plugins/ov_rest/ov_rest_re_discover.c	2025-01-16 09:34:34.914862068 +0100
@@ -704,7 +704,7 @@
 		byte bayNumber)
 {
 
-        SaErrorT rv = NULL;
+        SaErrorT rv;
         SaHpiResourceIdT resource_id = 0;
         struct oh_event event = {0};
 	struct ovRestHotswapState *hotswap_state = NULL;
