--- openhpi-2.10.2/plugins/ipmidirect/ipmi_con_lan.cpp.orig	2008-07-02 15:20:12.588941204 +0200
+++ openhpi-2.10.2/plugins/ipmidirect/ipmi_con_lan.cpp	2008-07-02 15:22:21.580075939 +0200
@@ -791,7 +791,7 @@
        tmsg[pos++] = 0x81; // Remote console IPMI Software ID
        tmsg[pos++] = r->m_seq << 2;
        tmsg[pos++] = eIpmiCmdSendMsg;
-       tmsg[pos++] =   r->m_send_addr.m_channel & 0xf
+       tmsg[pos++] = (r->m_send_addr.m_channel & 0xf)
                      | (1 << 6); // Turn on tracking
 
        if ( r->m_send_addr.m_type == eIpmiAddrTypeIpmbBroadcast )
--- openhpi-2.10.2/plugins/ipmidirect/ipmi_sensor_factors.cpp.orig	2008-07-02 15:25:20.067808883 +0200
+++ openhpi-2.10.2/plugins/ipmidirect/ipmi_sensor_factors.cpp	2008-07-02 15:25:53.065541025 +0200
@@ -421,10 +421,10 @@
             // If swap == true, when raw value increases
             // the corresponding interpreted value decreases
             // so we have to take that into account when searching
-            if ( swap == false )
+            if ( swap == false ) {
                if ((val > cval) && (raw < maxraw))
                     raw++;
-            else
+	    } else
                if ((val < cval) && (raw < maxraw))
                     raw++;
 
@@ -434,10 +434,10 @@
             // If swap == true, when raw value increases
             // the corresponding interpreted value decreases
             // so we have to take that into account when searching
-            if ( swap == false )
+            if ( swap == false ) {
                 if ( ( val < cval) && (raw > minraw ) )
                     raw--;
-            else
+	    } else
                 if ( ( val > cval) && (raw > minraw ) )
                     raw--;
 
