--- openhpi-2.12.0/acinclude.m4.orig	2008-07-25 19:24:26.000000000 +0200
+++ openhpi-2.12.0/acinclude.m4	2008-08-26 22:03:56.770123415 +0200
@@ -73,6 +73,8 @@
         OH_TYPE_SIZES=`./$OH_SSFILE`
         # feel free to define more logic here if we need it
     
+        oldifs="$IFS"
+        IFS=' '
         OH_SIZEOF_UCHAR=`echo -e $OH_TYPE_SIZES | grep "^unsigned char" | awk '{print $[3]}'`
         OH_SIZEOF_USHORT=`echo -e $OH_TYPE_SIZES | grep "^unsigned short" | awk '{print $[3]}'`
         OH_SIZEOF_UINT=`echo -e $OH_TYPE_SIZES | grep "^unsigned int" | awk '{print $[3]}'`
@@ -83,6 +85,7 @@
         OH_SIZEOF_FLOAT=`echo -e $OH_TYPE_SIZES | grep "^float" | awk '{print $[2]}'`
         OH_SIZEOF_DOUBLE=`echo -e $OH_TYPE_SIZES | grep "^double" | awk '{print $[2]}'`
         rm -f $OH_SSFILE $OH_SSSOURCE
+        IFS="$oldifs"
 
     fi
     ])
@@ -184,7 +187,7 @@
 				   OPENIPMI_VERSION_RELEASE);" >> $OH_OI_SRC
 	echo "return 0;}" >> $OH_OI_SRC
 
-	gcc -o $OH_OI_FILE $CFLAGS $CPPFLAGS $LDFLAGS $OH_OI_SRC >& /dev/null
+	gcc -o $OH_OI_FILE $CFLAGS $CPPFLAGS $LDFLAGS $OH_OI_SRC >/dev/null 2>&1
 
 	if test -f "ipmi_ver"; then
 		OPENIPMI_VERSION=`./ipmi_ver | \
