--- openhpi-3.0.0/transport/Makefile.am.orig	2011-10-17 17:57:27.000000000 +0200
+++ openhpi-3.0.0/transport/Makefile.am	2011-10-22 08:12:25.323358056 +0200
@@ -32,6 +32,7 @@
 	strmsock.h
 
 libopenhpitransport_la_LDFLAGS= -version-info @HPI_LIB_VERSION@
+libopenhpitransport_la_LIBADD= $(GLIB_ONLY_LIBS)
 
 clean-local:
 	rm -f *~ core core.*
--- openhpi-3.6.1/ssl/Makefile.am.orig	2015-09-15 00:48:33.000000000 +0200
+++ openhpi-3.6.1/ssl/Makefile.am	2015-12-15 21:26:02.249408789 +0100
@@ -19,5 +19,5 @@
 libopenhpi_ssl_la_SOURCES = oh_ssl.c oh_ssl.h
 
 libopenhpi_ssl_la_LDFLAGS = -version-info @HPI_LIB_VERSION@
-libopenhpi_ssl_la_LIBADD  = @CRYPTO_LIB@
+libopenhpi_ssl_la_LIBADD  = $(top_builddir)/utils/libopenhpiutils.la @CRYPTO_LIB@
 
