diff -urNp -x '*.orig' openhpi-3.8.0.org/plugins/ipmidirect/ipmi.cpp openhpi-3.8.0/plugins/ipmidirect/ipmi.cpp
--- openhpi-3.8.0.org/plugins/ipmidirect/ipmi.cpp	2018-03-08 19:10:50.000000000 +0100
+++ openhpi-3.8.0/plugins/ipmidirect/ipmi.cpp	2021-10-04 22:34:50.760393922 +0200
@@ -1950,7 +1950,8 @@ cIpmi::AllocConnection( GHashTable *hand
           }
 
        memcpy( &lan_addr, ent->h_addr_list[0], ent->h_length );
-       unsigned int a = *(unsigned int *)(void *)ent->h_addr_list[0];
+       unsigned int a;
+       memcpy(&a, ent->h_addr_list[0], sizeof(a));
 
        stdlog << "Using host at "
               << (int)(a & 0xff) << "."
