--- openhpi-2.1.1/Makefile.am.orig	2005-04-20 23:56:19.000000000 +0200
+++ openhpi-2.1.1/Makefile.am	2005-05-22 18:32:26.839041456 +0200
@@ -75,6 +75,8 @@
 include_HEADERS	= $(top_srcdir)/include/SaHpi.h \
 		  $(top_srcdir)/include/oHpi.h \
 		  $(top_srcdir)/utils/oh_utils.h \
+		  $(top_srcdir)/utils/el_utils.h \
+		  $(top_srcdir)/utils/epath_utils.h \
 		  $(top_srcdir)/utils/rpt_utils.h \
 		  $(top_srcdir)/utils/sahpi_enum_utils.h \
 		  $(top_srcdir)/utils/sahpi_event_encode.h \
