--- openhpi-0.7.0/plugins/ipmidirect/ipmi.cpp.orig	2004-02-04 03:15:11.000000000 +0000
+++ openhpi-0.7.0/plugins/ipmidirect/ipmi.cpp	2004-02-22 13:36:05.000000000 +0000
@@ -908,7 +908,8 @@
           }
 
        memcpy( &lan_addr, ent->h_addr_list[0], ent->h_length );
-       unsigned int a = *(unsigned int *)(void *)ent->h_addr_list[0];
+       unsigned int a;
+       memcpy(&a, ent->h_addr_list[0], sizeof(a));
 
        IpmiLog( "Using host at %d.%d.%d.%d.\n",
                 a & 0xff, (a >> 8 ) & 0xff, 
