diff -urNp -x '*.orig' openhpi-3.8.0.org/openhpid/Makefile.am openhpi-3.8.0/openhpid/Makefile.am
--- openhpi-3.8.0.org/openhpid/Makefile.am	2021-10-04 22:34:50.237088323 +0200
+++ openhpi-3.8.0/openhpid/Makefile.am	2021-10-04 22:34:50.380414060 +0200
@@ -82,9 +82,10 @@ openhpid_LDADD          += -lstdc++
 endif
 
 
+initddir=$(sysconfdir)/rc.d/init.d
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/init.d
-	$(INSTALL) -m 0755 openhpid.sh $(DESTDIR)$(sysconfdir)/init.d/openhpid
+	$(mkinstalldirs) $(DESTDIR)$(initddir)
+	$(INSTALL) -m 0755 openhpid.sh $(DESTDIR)$(initddir)/openhpid
 
 uninstall-local:
 	rm -f $(DESTDIR)$(sysconfdir)/init.d/openhpid
