--- openhpi-0.6.0/openhpi.pc.in.orig	2003-11-25 16:33:07.000000000 +0100
+++ openhpi-0.6.0/openhpi.pc.in	2004-03-27 19:11:49.635128520 +0100
@@ -6,6 +6,6 @@
 Name: OpenHPI
 Description: Implementation of SA Forum's Hardware Platform Interface
 Version: @VERSION@
-Requires: glib
+Requires: @GLIB@
 Libs: -L${libdir} -l@HPI_PKG@
 Cflags: -I${includedir}/@HPI_PKG@ 
--- openhpi-0.6.0/configure.ac.orig	2004-03-19 23:14:34.000000000 +0100
+++ openhpi-0.6.0/configure.ac	2004-03-27 19:15:09.014818200 +0100
@@ -88,6 +88,7 @@
 *** unsupported GLIB $GLIB_REQUIRED_VERSION. The latest version of
 *** GLIB is always available from ftp://ftp.gtk.org/.])
 esac
+AC_SUBST(GLIB)
 
 if pkg-config --atleast-version $GLIB_REQUIRED_VERSION $GLIB; then
    :
