--- openhpi-0.6.0/acinclude.m4.orig	2004-03-19 23:43:18.000000000 +0100
+++ openhpi-0.6.0/acinclude.m4	2004-03-27 13:05:57.859305696 +0100
@@ -41,7 +41,8 @@
 
     OH_TYPE_SIZES=`./$OH_SSFILE`
     # feel free to define more logic here if we need it
-    
+    oldifs="$IFS"
+    IFS=' '
     OH_SIZEOF_UCHAR=`echo -e $OH_TYPE_SIZES | grep "^unsigned char" | awk '{print $[3]}'`
     OH_SIZEOF_USHORT=`echo -e $OH_TYPE_SIZES | grep "^unsigned short" | awk '{print $[3]}'`
     OH_SIZEOF_UINT=`echo -e $OH_TYPE_SIZES | grep "^unsigned int" | awk '{print $[3]}'`
@@ -52,6 +53,7 @@
     OH_SIZEOF_FLOAT=`echo -e $OH_TYPE_SIZES | grep "^float" | awk '{print $[2]}'`
     OH_SIZEOF_DOUBLE=`echo -e $OH_TYPE_SIZES | grep "^double" | awk '{print $[2]}'`
     rm -f $OH_SSFILE $OH_SSSOURCE
+    IFS="$oldifs"
     ])
 
 #
