--- openhpi-0.6.0/plugins/ipmidirect/Makefile.am.orig	2004-03-11 09:54:12.000000000 +0100
+++ openhpi-0.6.0/plugins/ipmidirect/Makefile.am	2004-03-27 14:19:42.294689280 +0100
@@ -110,7 +110,6 @@
 
 plugin_test_SOURCES     = plugin_test.cpp
 plugin_test_LDADD       = libipmidirect.la $(top_builddir)/utils/libopenhpiutils.la -luuid -lm @SSL_LIB@
-plugin_test_LDFLAGS     = -static
 
 clean:
 	rm -f *~
