diff -urNp -x '*.orig' openhpi-3.8.0.org/acinclude.m4 openhpi-3.8.0/acinclude.m4
--- openhpi-3.8.0.org/acinclude.m4	2021-10-04 22:34:49.490461436 +0200
+++ openhpi-3.8.0/acinclude.m4	2021-10-04 22:34:49.630453960 +0200
@@ -236,12 +236,8 @@ AC_DEFUN([OH_CHECK_RTAS],
             rtas_activate_firmware();
         ],
         [
-	    if test -f "/usr/bin/lsvpd"; then
 		    have_rtas_lib=yes
         	    AC_MSG_RESULT(yes)
-	    else
-		    AC_MSG_RESULT(no)
-	    fi
        ],
        [AC_MSG_RESULT(no)]
     )])
