--- openhpi-0.7.0/openhpid/Makefile.am.orig	2004-05-04 01:13:06.000000000 +0200
+++ openhpi-0.7.0/openhpid/Makefile.am	2004-05-20 01:02:31.862036360 +0200
@@ -32,7 +32,6 @@
 	openhpid.cpp
 
 openhpid_LDADD          = $(top_builddir)/marshal/libmarshal.la $(top_builddir)/src/libopenhpi.la $(top_builddir)/utils/libopenhpiutils.la
-#openhpid_LDFLAGS        = -static
 
 # client library
 # please do not use any external libraries
@@ -54,7 +53,6 @@
 # check saHpiEventGet function of the daemon
 event_get_SOURCES     = event_get.c
 event_get_LDADD       = libopenhpiclient.la $(top_builddir)/marshal/libmarshal.la $(top_builddir)/utils/libopenhpiutils.la
-event_get_LDFLAGS     = -static
 
 
 clean:
--- openhpi-0.6.0/plugins/ipmidirect/Makefile.am.orig	2004-03-11 09:54:12.000000000 +0100
+++ openhpi-0.6.0/plugins/ipmidirect/Makefile.am	2004-03-27 14:19:42.294689280 +0100
@@ -110,7 +110,6 @@
 
 plugin_test_SOURCES     = plugin_test.cpp
 plugin_test_LDADD       = libipmidirect.la $(top_builddir)/utils/libopenhpiutils.la -luuid -lm @SSL_LIB@
-plugin_test_LDFLAGS     = -static
 
 clean:
 	rm -f *~
