--- openhpi-2.0.0/plugins/ipmi/ipmi.h.orig	2004-12-29 11:03:02.000000000 +0100
+++ openhpi-2.0.0/plugins/ipmi/ipmi.h	2005-01-20 20:26:37.916025800 +0100
@@ -25,16 +25,16 @@
 #include <unistd.h>
 
 #include <OpenIPMI/ipmiif.h>
-#include <OpenIPMI/ipmi_sel.h>
+#include <OpenIPMI/internal/ipmi_sel.h>
 #include <OpenIPMI/ipmi_smi.h>
 #include <OpenIPMI/ipmi_err.h>
 #include <OpenIPMI/ipmi_auth.h>
 #include <OpenIPMI/ipmi_lan.h>
 #include <OpenIPMI/selector.h>
-#include <OpenIPMI/ipmi_int.h>
+#include <OpenIPMI/internal/ipmi_int.h>
 #include <OpenIPMI/os_handler.h>
-#include <OpenIPMI/ipmi_domain.h>
-#include <OpenIPMI/ipmi_event.h>
+#include <OpenIPMI/internal/ipmi_domain.h>
+#include <OpenIPMI/internal/ipmi_event.h>
 #include <OpenIPMI/ipmi_posix.h>
 
 #include <SaHpi.h>
--- openhpi-2.0.0/plugins/ipmi/Makefile.am.orig	2005-01-20 19:47:48.915087000 +0100
+++ openhpi-2.0.0/plugins/ipmi/Makefile.am	2005-01-20 20:34:35.686393664 +0100
@@ -37,7 +37,7 @@
 
 INCLUDES		= @OPENHPI_INCLUDES@
 
-AM_CFLAGS		= -g @CFLAGS@ @OPENIPMI_CFLAGS@
+AM_CFLAGS		= @CFLAGS@ @OPENIPMI_CFLAGS@ -Wno-deprecated-declarations
 
 pkglib_LTLIBRARIES	= libipmi.la
 
--- openhpi-2.0.0/plugins/ipmi/ipmi_entity_event.c.orig	2004-12-31 23:01:18.000000000 +0100
+++ openhpi-2.0.0/plugins/ipmi/ipmi_entity_event.c	2005-01-20 21:05:29.909508904 +0100
@@ -187,7 +187,7 @@
 	    || (!strcmp(entry->ResourceTag.Data, "invalid"))
 	    || ((ipmi_entity_get_entity_id(entity) == 7 || 15))) {
 
-			char *str;
+			const char *str;
 			char *str2;
 
 			str = ipmi_entity_get_entity_id_string(entity);
