--- openhpi-0.6.0/openhpid/Makefile.am.orig	2004-03-15 10:00:51.000000000 +0100
+++ openhpi-0.6.0/openhpid/Makefile.am	2004-03-27 12:34:19.752861736 +0100
@@ -25,9 +25,9 @@
 EXTRA_DIST              = README openhpiclient.conf
 
 # marshal and connection used by the daemon and client library
-noinst_LIBRARIES        = libmarshal.a
+noinst_LTLIBRARIES        = libmarshal.la
 
-libmarshal_a_SOURCES    = \
+libmarshal_la_SOURCES    = \
 	connection.h \
 	connection.c \
 	marshal_hpi.h \
@@ -44,8 +44,7 @@
 	openhpid.h \
 	openhpid.cpp
 
-openhpid_LDADD          = libmarshal.a $(top_builddir)/src/libopenhpi.la
-openhpid_LDFLAGS        = -static
+openhpid_LDADD          = libmarshal.la $(top_builddir)/src/libopenhpi.la
 
 # client library
 # please do not use any external libraries
@@ -53,11 +52,11 @@
 lib_LTLIBRARIES	        = libopenhpiclient.la
 
 libopenhpiclient_la_SOURCES = \
-	${libmarshal_a_SOURCES} \
 	openhpiclient.c \
 	simple_config.h \
 	simple_config.c
 
+libopenhpiclient_la_LIBADD = libmarshal.la
 libopenhpiclient_la_LDFLAGS = -version-info @HPI_LIB_VERSION@
 
 clean:
--- openhpi-0.6.0/plugins/ipmidirect/Makefile.am.orig	2004-03-11 09:54:12.000000000 +0100
+++ openhpi-0.6.0/plugins/ipmidirect/Makefile.am	2004-03-27 14:19:42.294689280 +0100
@@ -110,7 +110,6 @@
 
 plugin_test_SOURCES     = plugin_test.cpp
 plugin_test_LDADD       = libipmidirect.la $(top_builddir)/utils/libopenhpiutils.la -luuid -lm @SSL_LIB@
-plugin_test_LDFLAGS     = -static
 
 clean:
 	rm -f *~
